contracts/FlowIDTableStaking.cdc

```diff
@@ -857,6 +857,70 @@ pub contract FlowIDTableStaking {
             }
         }
 
+        /// Enforces slot (count) limits for each node type (role). If more nodes
+        /// of a given type are registered than the limit, they are removed
+        /// randomly until the limits are met. When a node is removed, it's
+        /// committed tokens are transfered to it's unstaked bucket.
+        pub fun slotSelect() {
+            let allNodeIDs = FlowIDTableStaking.getNodeIDs()
+
+            let limits = [2, 2, 2, 2, 2]
+            var counts = [0, 0, 0, 0, 0]
+            for nodeID in allNodeIDs {
+                let nodeRecord = FlowIDTableStaking.borrowNodeRecord(nodeID)
+                if nodeRecord.initialWeight == 100 {
+                    counts[nodeRecord.role - 1] = counts[nodeRecord.role - 1] + 1
+                }
+            }
+
+            for index, element in counts {
```

Comment:
```suggestion
            for index, nodeCount in counts {
```

Reply:
As this loop will now be iterating through a dictionary, the "element" variable is no longer present. If it is included, the code still parses but the result is that the keys start at zero regardless of the fact that I coded them to start at one.

Reply:
yeah, the dictionary would change how you do this, so you can ignore this comment